src/api/records.tsx

```diff
@@ -109,18 +109,24 @@ export const useRecords = <T extends unknown = Record[]>(
         resultsPerPage: number;
         sort: SortType;
         dateRange: DateRange;
+        plotRecords: boolean;
       }
     ]
-  >
+  >,
+  plotRecords = false
 ): UseQueryResult<T, AxiosError> => {
   const { page, resultsPerPage, sort, dateRange } =
     useAppSelector(selectQueryParams);
   const { apiUrl } = useAppSelector(selectUrls);
 
   return useQuery(
-    ['records', { page, resultsPerPage, sort, dateRange }],
+    ['records', { page, resultsPerPage, sort, dateRange, plotRecords }],
```

Comment:
Hmm, thinking of this it could have potential negative side-effects. For example, if a user is plotting, then the user switches the page and opens a new plot - the plot will have to refetch the record data as the React query key has changed - even though the pagination is not relevant to the plotting. It may be that we should in fact separate the `useRecordsPaginated` and `useRecords` (not paginated) query functions. Thoughts? If there end up being more complex shared options than `onError` - we could easily extract those common options to share between them.

Reply:
I hadn't thought of that. So go back to two separate functions for useRecordsPaginated and useRecords (could rename this to useAllRecords or something, I don't know, just to make it clearer)?

Reply:
yeah, I think that's the best way as otherwise we could end up requerying lots of results in plotting.